app/controllers/application_controller.rb

```diff
@@ -0,0 +1,24 @@
+class ApplicationController < ActionController::Base
+
+  before_action :authorized
+  helper_method :current_user
+  helper_method :logged_in?
+  helper_method :usernameById
+
+  def current_user
+    User.find_by(id: session[:user_id])
+  end
+
+  def usernameById(id)
+    user = User.find_by(id: id)
+    user.username
+  end
+
```

Comment:
Good short methods!

Reply:
Maybe theres a way you can use .permit to require email,username and password so a user cannot pass something else in when signing up